src/main/kotlin/dk/cachet/carp/webservices/protocol/repository/CoreProtocolRepository.kt

```diff
@@ -205,4 +206,35 @@ class CoreProtocolRepository(
         return null
     }
 
+    /**
+     * The [getLatestProtocols] function sorts Protocols conditionally by latest Protocols.versionTag
+     * converts a [JsonNode] to a [GetLatestProtocolResponseDto].
+     *
+     * @return A [toLatestProtocolOverview] object containing the ProtocolOverviewDto.
+     */
+
+    suspend fun getLatestProtocols(): List<LatestProtocolOverview?>
+    {
+        val protocols = protocolRepository.findAll()
+        val latestProtocols = mutableListOf<LatestProtocolOverview?>()
+        for (i in protocols)
+        {
+            val protocolSnapshotId = (i.snapshot)!!.firstOrNull()
+            if (protocolSnapshotId != null)
```

Comment:
you can probably make this more readable or concise by using sg like `mapNotNull`

Reply:
okey